ocs_ci/helpers/helpers.py

```
@@ -4480,3 +4486,97 @@ def verify_log_exist_in_pod_logs(
+            f"jq -r '.data.\".dockerconfigjson\"|@base64d' > {pull_secret_path}",
+            shell=True,
+        )
+    exec_cmd(
+        f"oc image extract --registry-config {pull_secret_path} "
+        f"{image} --confirm "
+        f"--path {get_architecture_path(cli_type)}:{local_cli_dir}"
+    )
+    os.rename(
+        os.path.join(local_cli_dir, remote_cli_basename),
+        local_cli_path,
+    )
+    # Add an executable bit in order to allow usage of the binary
+    current_file_permissions = os.stat(local_cli_path)
+    os.chmod(
+        local_cli_path,
+        current_file_permissions.st_mode | stat.S_IEXEC,
+    )
+    # Make sure the binary was copied properly and has the correct permissions
+    assert os.path.isfile(
+        local_cli_path
+    ), f"{cli_type} CLI file not found at {local_cli_path}"
+    assert os.access(
+        local_cli_path, os.X_OK
+    ), f"The {cli_type} CLI binary does not have execution permissions"
+
+
+def get_architecture_path(cli_type):
+    """
+    Get Architcture path
+
+    Args:
+        cli_type (str): choose which bin file you want to download ["odf" -> odf-cli , "mcg" -> mcg-cli]
+
+    Returns:
+        (str): path of MCG/ODF CLI Binary in the image.
+    """
+    system = platform.system()
+    machine = platform.machine()
+    path = f"/usr/share/{cli_type}/"
+    if cli_type == "mcg":
+        image_prefix = "noobaa"
+    elif cli_type == "odf":
+        image_prefix = "odf"
+    if system == "Linux":
+        path = os.path.join(path, "linux")
+        if machine == "x86_64":
+            path = os.path.join(path, f"{image_prefix}-amd64")
+        elif machine == "ppc64le":
+            path = os.path.join(path, f"{image_prefix}-ppc64le")
+        elif machine == "s390x":
+            path = os.path.join(path, f"{image_prefix}-s390x")
+    elif system == "Darwin":  # Mac
+        path = os.path.join(path, "macosx", image_prefix)
```

Check with Boris about M2/M1 chip (`arm64`) supported as I remember they build only for x86_64 mac on Intel chip.,  In that case we can throw an exception for in case of machine == "arm64" as not supported platform.

---

I've tested this on my own M1 Mac and asked several other ODF-QE members with M1 Macs to test this as well, so I can confirm that since 4.14 the build under "Darwin" here is also supported for M1 chips.

This is why I removed the original arm64 architecture check here: https://github.com/red-hat-storage/ocs-ci/pull/8155

---

Great, that I didn't know